### PR TITLE
Update Curriculum Visualizer Tab

### PR DESCRIPTION
**Pull request by lioneljingles**
## Description

- Added UI elements to visualize curriculum units in curriculum `Unit Sequence` tab
- Integrated Curriculum 2023 API for unit data
- Updated subject phase picker to pre-select for current view
- Deleted legacy subjects from subject phase picker

## How to test

1. Go to https://deploy-preview-1798--oak-web-application.netlify.thenational.academy/beta/teachers/curriculum/maths-secondary/units
2. You should see a grid of the units in the maths secondary curriculum sequence
3. For years 
4. You should see "Maths" and "Secondary" selected in the subject phase picker
5. In the subject phase picker, click on "Geography" then "Primary"
6. You should see a grid of the units in the geography primary curriculum sequence
7. Go to https://deploy-preview-1798--oak-web-application.netlify.thenational.academy/beta/teachers/curriculum/science-secondary/units
8. For year 10 and year 11, you should see filters for child subjects and tiers
9. Click "Physics" subject and "Higher" tier
10. You should see the units updated accordingly
11. Go to: https://deploy-preview-1798--oak-web-application.netlify.thenational.academy/beta/teachers/curriculum/english-primary/units
12. You should see domains for all years
13. Click "Grammar" for year 1
14. You should see the units updated accordingly

## Checklist

- [x] Added or updated tests where appropriate
- [x] Manually tested across browsers / devices
- [x] Considered impact on accessibility
- [x] Design sign-off
- [x] Approved by product owner

### CODE REVIEW

File: src/components/pages/CurriculumInfo/CurriculumHeader/CurriculumHeader.tsx [hunk 1]
```diff
@@ -91,27 +128,28 @@ const CurriculumHeader: FC<CurriculumHeaderPageProps> = ({
               label: "Overview",
               page: "curriculum-overview",
               viewType: "teachers",
-              subjectPhaseSlug: pageSlug,
+              subjectPhaseSlug: subjectPhaseSlug,
               isCurrent: tab == "overview",
```
Comment (k-huggs) [inline src/components/pages/CurriculumInfo/CurriculumHeader/CurriculumHeader.tsx:132]: ===

File: src/node-lib/curriculum-api-2023/queries/curriculumOverview/curriculumOverview.query.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import curriculumOverviewTabFixture from "@/node-lib/curriculum-api-2023/fixtures/curriculumOverview.fixture";
+
+describe("curriculum overview query", () => {
+  test("throws params incorrect error if slugs are blank", async () => {
+    expect(async () => {
+      await curriculumOverviewQuery()({
+        // {
+        //   ...sdk,
+        //  curriculumOverview: jest.fn(() => Promise.resolve({ data: {} })),
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/queries/curriculumOverview/curriculumOverview.query.test.ts:12]: delte comments

**Review by rightsaidjames — APPROVED** [review]
Approved from a functional/quality perspective, notwithstanding the data accuracy and URL issues flagged elsewhere. Obviously I'll leave the code approval to @k-huggs 😅

**Review by HelenHarlow — CHANGES_REQUESTED** [review]
Some small UI things noted:

- The pink blocks that contain the unit cards should have a corner radius of 4px
- The unit card component in design has a background shape with rough edges. Is it possible to add this into the component? (I know sometimes these are tricky to work with)
-  In the mobile journey, in design, the tabs component for the filters becomes a drop down to avoid them running over 2 lines. I’m not worried about this for now as I know we’re not optimising mobile, but is it possible to add some vertical padding, as they overlap when then run over 2 lines?

Otherwise, looks great!

File: src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx [hunk 3]
```diff
@@ -1,94 +1,339 @@
...
+  });
+
+  const initialYearSelection = {} as YearSelection;
+  Object.keys(yearData).forEach((year) => {
+    const data = yearData[year];
+    if (!data) {
+      throw new Error("year data missing");
+    }
+    data.childSubjects.sort((a, b) => {
+      if (a.subject_slug === "combined-science") {
```
Comment (mantagen) [inline src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:97]: Is there a way this sort can be achieved on the backend? i.e. not referencing a specific subject slug in this codebase?

File: src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx [hunk 4]
```diff
@@ -1,94 +1,339 @@
...
+  const yearData: {
+    [key: string]: {
+      units: Unit[];
+      childSubjects: Subject[];
+      domains: Domain[];
+      tiers: Tier[];
+    };
+  } = {};
+
+  data.units.forEach((unit) => {
```
Comment (mantagen) [inline src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:52]: Seems like there's a fair amount of wrangling with data inline in this component -- would be great to the functionality it tested ... could pull it out into its own function?

**Review by mantagen — COMMENTED** [review]
Looking great! Just the point about combined-science, and the other about inline data manipulation.

Neither blockers ... I know we're up against it at the moment

File: src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx (hunk 3 above)
Comment (lioneljingles) [reply src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:97]: Good point, will address in the MV and create a different PR for it if that's alright

**Comment by lioneljingles** [comment]
> Some small UI things noted:
> 
>     * The pink blocks that contain the unit cards should have a corner radius of 4px
> 
>     * The unit card component in design has a background shape with rough edges. Is it possible to add this into the component? (I know sometimes these are tricky to work with)
> 
>     * In the mobile journey, in design, the tabs component for the filters becomes a drop down to avoid them running over 2 lines. I’m not worried about this for now as I know we’re not optimising mobile, but is it possible to add some vertical padding, as they overlap when then run over 2 lines?
> 
> 
> Otherwise, looks great!

Made changes in latest commit. There are some issues getting the spacing correct with the rough edges background, but should be an improvement!

**Comment by rightsaidjames** [comment]
I've manually confirmed:
* 4px border radius for pink containers
* Rough borders for unit card components
* Tab filters (child subject, domain) wrap onto
 two lines.

My only query with the latter (@HelenHarlow) is whether there should be some extra padding below the tabs when there is no tier filter? (See third screenshot)

Screenshot 2023-09-07 at 15 44 10
Screenshot 2023-09-07 at 15 45 00
Screenshot 2023-09-07 at 15 48 50

**Comment by HelenHarlow** [comment]
> I've manually confirmed:
> 
> * 4px border radius for pink containers
> * Rough borders for unit card components
> * Tab filters (child subject, domain) wrap onto
>   two lines.
> 
> My only query with the latter (@HelenHarlow) is whether there should be some extra padding below the tabs when there is no tier filter? (See third screenshot)
> 
> Screenshot 2023-09-07 at 15 44 10 Screenshot 2023-09-07 at 15 45 00 Screenshot 2023-09-07 at 15 48 50

Amazing that looks great now! Yes that does look a little tight, in Figma I have that gap as 16px

**Review by HelenHarlow — APPROVED** [review]

**Pull request merged**